int32 loc
int32 belief